backend/src/grynos.js

```diff
@@ -27,26 +27,30 @@ const mapCourseFromGrynos = (course) => ({
 });
 
 const fetchCourses = async () => {
-    const [response] = await Promise.all([
-        axios(url),
-        db.sync({ force: true }),
-    ]);
-    return response.data.course.map(mapCourseFromGrynos);
+    const response = await axios(url);
+    if (response.data.course) {
+        await db.sync({ force: true });
+        return response.data.course.map(mapCourseFromGrynos);
+    }
 };
 
-const updateCoursesToDb = () => {
+const updateCoursesToDb = async () => {
     try {
-        const courses = fetchCourses();
-        courses.forEach((course) => {
-            models.courses.create(course, {
-                include: [
-                    { model: models.locations, as: 'location' },
-                    { model: models.events, as: 'teachingSession' },
-                ],
+        const courses = await fetchCourses();
+        if (courses) {
```

Comment:
I think this should be refactored to wait also for the course creation? So instead of using `forEach` it would be `return await Promise.all(courses.map(...))`.